src/main/java/com/google/cloud/tools/appengine/configuration/RunConfiguration.java

```diff
@@ -354,7 +354,9 @@ public Builder pythonStartupArgs(@Nullable String pythonStartupArgs) {
     }
 
     public Builder jvmFlags(@Nullable List<String> jvmFlags) {
```

Comment:
`jvmFlags(null)` will be a no-op, and it may feel confusing. I think just taking out `@Nullable` so that people can't call `jvmFlags(null)`? The down side is that we pass the null check down to the clients. Also it may not easy to find places in client code where null can get passed.

But is returning a mutable internal state is the only best way? Can't we have like `addJvmFlags()`?

Reply:
Or, maybe passing around the builder like @briandealwis said?

Reply:
It's not the only way. It's the way with least code change, here and in Eclipse. Passing a builder really doesn't look right in Eclipse, and it's not a safe change. Note that right now this method does expose mutable internal state. This PR makes that explicit.

Te alternative is adding a toBuilder method that converts a RunConfiguration back into a Builder. It's a lot more code though.